hadoop-ozone/s3gateway/src/main/java/org/apache/hadoop/ozone/s3/endpoint/EndpointBase.java

```diff
@@ -190,6 +202,38 @@ public void deleteS3Bucket(String s3BucketName)
     }
   }
 
+  protected Map<String, String> getCustomMetadataFromHeaders(
```

Comment:
We should apply limits 
```
 Within the PUT request header, the user-defined metadata is limited to 2 KB in size. The size of user-defined metadata is measured by taking the sum of the number of bytes in the UTF-8 encoding of each key and value. ```
https://docs.aws.amazon.com/AmazonS3/latest/userguide/UsingMetadata.html

Reply:
I just checked and Aravindan had addressed it.